data/lua/explode_ability.lua

```diff
@@ -0,0 +1,38 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
+	local ec = wesnoth.current.event_context
+
+	if not ec.x1 or not ec.y1 then
+		return
+	end
+
+	local u_exploder = wesnoth.units.get(ec.x1, ec.y1)
+
+	if not u_exploder or not u_exploder:matches { ability = "explode" } then
+		return
+	end
+
+	local u_exploder_cfg = u_exploder.__cfg
+        local adjacent_units = wesnoth.units.find_on_map{T.filter_adjacent { id = u_exploder_cfg.id}}
+
+	for i,v in ipairs(wml.get_child(u_exploder_cfg, "abilities")) do
```

Comment:
I think it's better to get rid of the extra variable here:
```suggestion
        local adjacent_units = wesnoth.units.find_on_map{T.filter_adjacent { id = u_exploder.id}}

	for i,v in ipairs(wml.get_child(u_exploder.__cfg, "abilities")) do
```

Also, it looks like you need to run the whitespace fixing script. If you're not on Windows, you can just run the script `./utils/CI/fix_whitespace.sh`. I'm not sure what options there are if you _are_ on Windows, other than running that in `git bash` or something.